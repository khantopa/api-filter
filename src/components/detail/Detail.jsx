import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import axios from 'axios';
import './Detail.scss';

export default class Detail extends Component {
  constructor(props) {
    super(props)
    this.state = {
      contact: {},
      company: {}
    }
  }

  componentDidMount = () => {
    const id = this.props.match.params.id;
    axios.get(`http://165.22.60.22/api/v1/contacts/${id}`)
    .then(response => {
      this.setState({ contact: response.data.data, company: response.data.data.company })
    })
    .catch( err => console.log(err))
  }


  render() {
    const { name, email } = this.state.contact;
    const company = this.state.company;
    return (
      <div className="contact">
        <Link to='/contacts' className="button">Back</Link>
        <div className="contact_card">
          <h1>{ name }</h1>
          <h6>E-mail: { email }</h6>
          <div className="company">
            <p><span>Company ID:</span>&nbsp;&nbsp; { company.id }</p>
            <p><span>Company:</span>&nbsp;&nbsp; { company.name }</p>
            <p><span>Revenue:</span>&nbsp;&nbsp; { company.revenue }</p>
            <p><span>Country:</span>&nbsp;&nbsp; { company.country }</p>
          </div>
        </div>
      </div>
    )
  }
}
