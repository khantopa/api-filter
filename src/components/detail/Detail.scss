@import '../../styles/main.scss';

.contact {
  position: relative;
  height: 100vh;
  width: 100%;
  padding: 40px;
  box-sizing: border-box;
  background: $primary;
  .button {
    width: 100px;
    color: $light;
    background: $dark;
  }
  &_card {
    @include absPosition(50%, 50%);
    transform: translate(-50%, -50%);
    @include card(80vh, auto);
    background: rgba(255, 255, 255, .9);
    color: $primary;
    h1 {
      text-transform: uppercase;
      line-height: 38px;
      font-size: 30px;
      letter-spacing: .2em;
    }
    h6 {
      font-size: 14px;
      color: $dark;
    }
    .company {
      margin: 30px auto;
      text-align: left;
      p {
        font-size: 18px;
        color: $dark;
        span {
          font-weight: bold;
          color: $primary;
        }
      }
    }
  }
}

@media (max-width: 700px) {
  .contact {
    &_card {
      position: unset;
      transform: unset;
      @include card(100%, auto);
    }
  }
}