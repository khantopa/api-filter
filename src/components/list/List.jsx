import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import axios from 'axios';
import './List.scss';

import Search from '../search/Search';

export default class List extends Component {
  constructor(props) {
    super(props);
    this.state = {
      lists: []
    }
  }

  

  onNameChange = async(name) => {
    axios.get(`http://165.22.60.22/api/v1/contacts?name=${name}`)
    .then(response => {
      if(response.data.data !== null) {
        this.setState({lists: response.data.data})
      } else {
        this.getAllData();
      }
    })
  }

  onCompanyChange = async(id) => {
    axios.get(`http://165.22.60.22/api/v1/contacts?company_id=${id}`)
    .then(response => {
      if(response.data.data !== null) {
        this.setState({lists: response.data.data})
      } else {
        this.getAllData();
      }
    })
  }

  onRevenueChange = async(value) => {
    axios.get(`http://165.22.60.22/api/v1/contacts?revenue_gte=${value}`)
    .then(response => {
      if(response.data.data !== null) {
        this.setState({lists: response.data.data})
      } else {
        this.getAllData();
      }
    })
  }

  componentDidMount = () => {
    this.getAllData();
  }

  getAllData = () => {
    axios.get('http://165.22.60.22/api/v1/contacts')
    .then(response => {
      this.setState({lists: response.data.data})
    })
    .catch(err => 'err');
  }

  render() {

    const tableRows = this.state.lists.map( row => (
      <tr key={row.id} className="table_body_row">
        <td className="table_body_row_col">
          <Link to={`/contacts/${row.id}`}>{row.name}</Link>
        </td>
        <td className="table_body_row_col">{ row.company.name }</td>
        <td className="table_body_row_col">{ row.company.revenue }</td>
      </tr>
    ));

    const table = (
      <table className="table">
        <thead className="table_head">
          <tr className="table_head_row">
            <td className="table_head_row_col">Name</td>
            <td className="table_head_row_col">Company</td>
            <td className="table_head_row_col">Revenue</td>
          </tr>
        </thead>
        <tbody className="table_body">
          { tableRows }
        </tbody>
      </table>
    );


    return (
      <div className="container">
        <h1>Lists</h1>
        <br/>
        <div className="filter">
          <Search label="Search by name" onInputChange={ this.onNameChange } />
          <Search label="Search by company id" onInputChange={ this.onCompanyChange }/>
          <Search label="Search by revenue" onInputChange={ this.onRevenueChange }/>
        </div>
        { table }
      </div>
    )
  }
}
