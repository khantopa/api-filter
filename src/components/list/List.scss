@import '../../styles/main.scss';

.container {
  width: 100%;
  padding: 40px 5%;
  .filter {
    @include grid(3);
    grid-gap: 30px;
    margin-bottom: 20px;
  }
  .table {
   @include card(100%, auto);
    box-sizing: border-box;
    padding: 20px;
    box-shadow: 0 5px 10px 5px rgba(0, 0, 0, .15);
    &_head {
      width: 100%;
      background: $dark;
      color: $light;
      height: 40px;
      font-weight: bold;
    }
    &_body {
      &_row {
        height: 40px;
        border-bottom: 1px solid $light;
        &:hover {
          background: $light;
        }
        a {
          text-decoration: none;
          color: $primary;
          font-weight: bold;
          &:hover {
            color: $dark;
          }
        }
      }
    }
  }
}

