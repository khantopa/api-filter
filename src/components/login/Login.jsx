import React, { Component } from 'react';
import User from'../../images/user-white.svg';
import Email from '../../images/email.svg';
import Key from '../../images/key.svg'
import './Login.scss';


export default class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      status: '',
      isLogedIn: false
    }
  }

  handleChange = (e) => {
    let name = e.target.name;
    let value = e.target.value
    this.setState({
      [name]:value,
      status: ''
    })
  }

  login = () => {
    const email = this.state.email;
    const password = this.state.password;
    if(email === "test@test.com" && password === "12345678") {
      this.props.history.push('/contacts');
    } else {
      this.setState({ status: 'Incorrect username and password'})
    }
  }

  render() {
    return (
      <div className="login">
        <div className="login_card">
          <div className="login_icon">
            <img className="login_icon_img" src={User} alt="User-icon" />
          </div>
          <div className="login_form">
            <label>
              <input type="email" className="login_form_field" placeholder="E-mail" 
              name="email" onChange={ this.handleChange } id="email" /> 
              <img className="icon" src={ Email } alt="Email" /> 
            </label>
            <label>
              <input type="password" className="login_form_field" placeholder="Password" 
              name="password" onChange={ this.handleChange } id="password" />
              <img className="icon" src={ Key } alt="Key" />
            </label>
          </div>
          { this.state.status !== ''
            ? <div className="error" style={{'background': 'red'}}>
                <p>{ this.state.status }</p>
              </div>
            : <div className="error"></div>
          }
          <br/><br/>
          <div className="login_submit">
            <button onClick={ this.login } className="login_submit_button">Login</button>
          </div>
        </div>
      </div>
    )
  }
}
