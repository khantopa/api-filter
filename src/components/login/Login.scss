@import '../../styles/main.scss';

.login {
  position: relative;
  width: 100vw;
  height: 100vh;
  overflow: hidden;
  background: url('../../images/bg.jpg') center/cover no-repeat;
  color: $light;

  &_icon {
    @include grid(1);
    place-items: center;
    &_img {
      height: 80px;
      width: 80px;
    }
  }
  &_card {
    @include absPosition(50%, 50%);
    @include card(320px, auto);
    transform: translate(-50%, -50%);
  }

  &_form {
    padding: 30px 0 10px 0;
    @include grid(1);
    grid-gap: 20px;
    &_field {
      @extend .form_field;
      text-align: left;
      padding-left: 40px;
    }
    label {
      position: relative;
      .icon {        
        @include absPosition(4%, 50%);
        transform: translateY(-50%);
        width: 20px;
      }
    }
  }
  &_submit {
    @include grid(1);
    &_button {
      @extend .button;
      width: 40%;
      background: $dark;
    }
  }
} 