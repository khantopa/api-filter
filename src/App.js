import React from 'react';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import './App.scss';

import Login from './components/login/Login';
import List from './components/list/List';
import Detail from './components/detail/Detail';

const App = () => {
  return (
    <div className ="App">
      <Router>
        <Route exact path='/' component={ Login }></Route>
        <Route exact path='/contacts' component= { List }></Route>
        <Route path='/contacts/:id' component={ Detail }></Route>
      </Router>
    </div>
  )
}

export default App;
