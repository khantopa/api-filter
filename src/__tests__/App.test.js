import ReactDOM from 'react-dom';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import App from '../App';

import Login from '../components/login/Login';
import List from '../components/list/List';
import Detail from '../components/detail/Detail';

let wrapper, router;

beforeEach(() => {
  wrapper = shallow(<App/> );
  router = wrapper.find(Router);
});

afterEach(() => {
  wrapper.unmount();
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});


it('should render 1 Router', () => {
  expect(router).toHaveLength(1);
});

it('should rednder 3 Routes', () => {
  expect(router.find(Route)).toHaveLength(3);
});

it('Should renders correct routes', () => {
  const pathMap = router.find(Route).reduce((pathMap, route) => {
    const routeProps = route.props();
    pathMap[routeProps.path] = routeProps.component;
    return pathMap;
  }, {});  
  expect(pathMap['/']).toBe(Login);
  expect(pathMap['/contacts']).toBe(List);
  expect(pathMap['/contacts/:id']).toBe(Detail);
});
