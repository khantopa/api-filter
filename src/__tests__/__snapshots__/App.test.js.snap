// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="App"
>
  <BrowserRouter>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/contacts"
    />
    <Route
      component={[Function]}
      path="/contacts/:id"
    />
  </BrowserRouter>
</div>
`;
